qa/tasks/vstart_runner.py

```diff
@@ -175,62 +175,130 @@ def __init__(self, args, subproc, check_status, stdout, stderr):
         if self.finished:
+            log.info('self.finished is true')
             # Avoid calling communicate() on a dead process because it'll
             # give you stick about std* already being closed
             if self.check_status and self.exitstatus != 0:
+                log.info('self.finished is True')
+                log.info('self.args = ' + str(self.args))
+                log.info('self.exitstatus = ' + str(self.exitstatus))
+                log.info('raising CommandFailedError')
                 raise CommandFailedError(self.args, self.exitstatus)
             else:
+                log.info('finished running LocalRemoteProcess.wait(), returning None')
                 return
+        log.info('self.finished is False, running self.subproc.communicate '
+                 'next')
 
         out, err = self.subproc.communicate()
+        log.info('finished running self.subproc.communicate(), delete nonascii chars next')
         out, err = rm_nonascii_chars(out), rm_nonascii_chars(err)
+        log.info('finished deleting nonascii chars')
         if isinstance(self.stdout, StringIO):
+            log.info('writing out to self.stdout after decoding...')
             self.stdout.write(out.decode(errors='ignore'))
         elif self.stdout is None:
             pass
         else:
+            log.info('writing out to self.stdout without decoding...')
             self.stdout.write(out)
         if isinstance(self.stderr, StringIO):
+            log.info('writing err to self.stderr after decoding...')
             self.stderr.write(err.decode(errors='ignore'))
         elif self.stderr is None:
             pass
         else:
+            log.info('writing err to self.stderr without decoding...')
             self.stderr.write(err)
 
         self.exitstatus = self.returncode = self.subproc.returncode
+        log.info('self.existatus = ' + str(self.exitstatus))
 
         if self.exitstatus != 0:
             sys.stderr.write(out.decode())
             sys.stderr.write(err.decode())
 
         if self.check_status and self.exitstatus != 0:
+            log.info('self.args = ' + str(self.args))
+            log.info('raising CommandFailedErorr')
             raise CommandFailedError(self.args, self.exitstatus)
+        log.info('finished running LocalRemoteProcess.wait(), returning None')
 
     @property
     def finished(self):
+        log.info('==================running LocalRemoteProcess.finished()')
         if self.exitstatus is not None:
+            log.info('self.existatus is not None')
+            log.info('finished running LocalRemote.finsished(), returning True')
             return True
+        log.info('self.exitstatus is None, running self.subproc.poll() next')
 
         if self.subproc.poll() is not None:
+            log.info('finished running self.subproc.poll(), its not None')
+            log.info('running self.subproc.communicate()')
             out, err = self.subproc.communicate()
```

Comment:
so it hangs here? what I noticed in the control flow is that you enver actually closed stdout/stderr:

````
2020-12-18 06:35:09,162.162 INFO:__main__:=============runinng LocalRemoteProcess.close_stdout_and_stderr. self.args = ['./bin/ceph', '-w']
2020-12-18 06:35:09,163.163 INFO:__main__:=============finished returning close_stdout_and_stderr
2020-12-18 06:35:09,163.163 INFO:tasks.ceph_test_case:=============finished closing self.watcher_process.stdout and stderr
````

it never hit any of the debug points where stdout/stderr are closed

Reply:
Yes. I spoke about this during our meeting, `self.subproc.poll()` returns when  None called in `LocalRemoteProcess.close_stdout_and_stderr()` but not in `LocalRemoteProcess.finished()`, which is odd because call to `LocalRemoteProcess.close_stdout_and_stderr()` precedes `LocalRemotProcess.finished()`.

Reply:
@batrick Did you get a chance to take a look into this PR?

Reply:
We can discuss this sometime after Pacific is released.